src/ai_hawk/llm/llm_manager.py

```diff
@@ -730,8 +730,14 @@ def is_job_suitable(self):
         )
         output = self._clean_llm_output(raw_output)
         logger.debug(f"Job suitability output: {output}")
-        score = re.search(r"Score:\s*(\d+)", output, re.IGNORECASE).group(1)
-        reasoning = re.search(r"Reasoning:\s*(.+)", output, re.IGNORECASE | re.DOTALL).group(1)
+
+        try:
+            score = re.search(r"Score:\s*(\d+)", output, re.IGNORECASE).group(1)
```

Comment:
re.search(r"Score:\s*(\d+)", output, re.IGNORECASE) will be none

error will be can't find attribute group(1) for oject type none.

Reply:
I don't understand what you mean. Are you suggesting I remove the .group(1) part at the end? It shouldn't be removed. 

Here's how this current implementation work:
try to find score & reasoning using regex, if it couldnt parse the response, the search will be of NoneType and we will be trying to do group(1) on a NoneType and will be caught with the AttributeError exception. I have tested this and it works as expected. 

If you remove the .group(1) from the try block, it wont throw any errors because if it couldnt get the score/reason, the search will just be of None value but that wont throw any exception.

Reply:
sorry my bad. understood. good work @49Simon